google-cloud-firestore/Gemfile

```diff
@@ -6,6 +6,7 @@ gem "google-cloud-core", path: "../google-cloud-core"
 gem "google-cloud-errors", path: "../google-cloud-errors"
 gem "google-cloud-firestore-v1", path: "../google-cloud-firestore-v1"
 
+gem "algorithms"
```

Comment:
Seems like this gem is not maintained actively, last commit is on 2021. Can we check if there is an alternative and also a small benchmarking to verify if using heap is really being helpful for this use case ?